inst/www/htmlwidgets.js

```diff
@@ -233,7 +233,21 @@
           theseArgs = theseArgs.concat([task.data]);
           task = task.code;
         }
-        var taskFunc = eval("(" + task + ")");
+        var taskFunc = null;
+        try {
+          taskFunc = eval(task);
+        } catch(error) {
+          if (!error instanceof SyntaxError) {
+            throw new Error(error.message);
```

Comment:
Try `throw error;` instead, so we preserve the class and other properties (and stack trace too, I believe?).